packages/mds-service-helpers/@types/index.ts

```diff
@@ -48,7 +48,13 @@ export type ServiceClient<S> = {
 }
 
 export type ServiceProvider<S> = {
-  [M in keyof S]: S[M] extends AnyFunction<infer R> ? (...args: Parameters<S[M]>) => Promise<ServiceResponse<R>> : never
+  [M in keyof S]: S[M] extends (...args: infer P) => infer R
+    ? (
+        ...args: {
+          [K in keyof P]: undefined extends P[K] ? Nullable<P[K]> : P[K]
+        }
+      ) => Promise<ServiceResponse<R>>
+    : never
```

Comment:
If a service client method has an optional parameter and the caller does not supply a value, `undefined` gets converted to `null` when the request is sent over the wire. The above `ServiceProvider` type change forces the corresponding service provider method to specifically account for the possibility of receiving `null`,